scrapligo/v2 panics on Linux when poll fd exceeds select/FdSet limit

# Problem
Under heavy concurrent `Cli.Open()` calls on Linux, scrapligo panics in `cli.getResult` when the `libscrapli` poll fd exceeds the `unix.Select`/`unix.FdSet` limit of 1023.
## Panic
```
panic: runtime error: index out of range [16] with length 16
goroutine 141 [running]:
runtime/debug.Stack()
	/opt/homebrew/Cellar/go/1.26.3/libexec/src/runtime/debug/stack.go:26 +0x5e
main.worker.func1.1()
	/Users/dean/develop/network-commander/cmd/loadtest/main.go:252 +0x12b
panic({0x77b220?, 0x1cba687391a0?})
	/opt/homebrew/Cellar/go/1.26.3/libexec/src/runtime/panic.go:860 +0x13a
golang.org/x/sys/unix.(*FdSet).Set(...)
	/Users/dean/go/pkg/mod/golang.org/x/sys@v0.43.0/unix/fdset.go:11
github.com/scrapli/scrapligo/v2/cli.(*Cli).getResult(0x1cba688817c0, {0x7d2d38, 0x1cba68bc1ce0}, 0x1cba68cd5a64, 0x1)
	/Users/dean/go/pkg/mod/github.com/scrapli/scrapligo/v2@v2.0.0-rc.10/cli/cli.go:303 +0xb85
github.com/scrapli/scrapligo/v2/cli.(*Cli).Open(0x1cba688817c0, {0x7d2d38, 0x1cba68bc1ce0})
	/Users/dean/go/pkg/mod/github.com/scrapli/scrapligo/v2@v2.0.0-rc.10/cli/cli.go:231 +0x1bb
main.worker.func1(0x1cba6879c0e0, 0x1cba688ebef8, {0x7e3?, 0x2?}, 0x77, 0x1cba687922e0, {0x7d2cc8, 0x1cba68836000})
	/Users/dean/develop/network-commander/cmd/loadtest/main.go:315 +0x80a
main.worker({0x7d2cc8, 0x1cba68836000}, 0x77, {{0x7ffc62ba5655, 0xc}, {0x7ffc62ba5645, 0x9}, {0x79eab6, 0x5}, {0x79eab6, ...}, ...}, ...)
	/Users/dean/develop/network-commander/cmd/loadtest/main.go:423 +0xaf
created by main.main in goroutine 1
	/Users/dean/develop/network-commander/cmd/loadtest/main.go:159 +0x81d
```
# Version
- **scrapligo**: `v2.0.0-rc.10`
- **libscrapli**: `v0.0.1-rc.14` 
# Environment
- **Go**: `1.26.3`
- **Platform**: `Linux/amd64`
# Expected
- Cli.Open() should not panic when the process has high-numbered file descriptors
# Actual
- process panics in unix.FdSet.Set during Cli.Open() result polling
# Steps to reproduce
```
# setup cisshgo (https://github.com/tbotnz/cisshgo) to accept connections on multiple ports 
$ docker run --rm --name cisshgo-loadtest --network clab --ip 172.20.20.10 cisshgo -l 100 -p 2000

# use the vibecoded loadtest tool from https://gist.github.com/euphdk/1c4d7c843e64fc4aaaf4730128208436
$ ./loadtest-linux-amd64 -ports 2000-2099 -host 172.20.20.10 -iterations 1000 -open-timeout 30s -command-timeout 30s -concurrency 400 
```

--- a/cli/cli.go
+++ b/cli/cli.go
@@ -308,23 +308,18 @@ func (c *Cli) getResult(
 
 	var n int
 
-	readFdSet := &unix.FdSet{}
+	pollFds := []unix.PollFd{{Fd: int32(c.pollFd), Events: unix.POLLIN}}
 
 	for {
 		if ctx.Err() != nil {
 			return nil, ctx.Err()
 		}
 
-		// select can modify both the fd set and the timeout (on some platforms), so we must
-		// re-initialize them both every iteration.
-		tv := unix.NsecToTimeval(scrapligoconstants.ReadyFDPollTimeoutNs)
-
-		readFdSet.Zero()
-		readFdSet.Set(c.pollFd)
+		pollFds[0].Revents = 0
 
 		var err error
 
-		n, err = unix.Select(c.pollFd+1, readFdSet, nil, nil, &tv)
+		n, err = unix.Poll(pollFds, scrapligoconstants.ReadyFDPollTimeoutMs)
 		if err != nil {
 			if errors.Is(err, unix.EINTR) {
 				// python automagically handles interrupts i guess go doesnt, so just act like
@@ -336,6 +331,13 @@ func (c *Cli) getResult(
 		}
 
 		if n > 0 {
+			if pollFds[0].Revents&unix.POLLNVAL != 0 {
+				return nil, scrapligoerrors.NewFfiError(
+					"waiting on operation ready signal",
+					unix.EBADF,
+				)
+			}
+
 			break
 		}
 	}

--- a/cli/cli.go
+++ b/cli/cli.go
@@ -308,7 +308,7 @@ func (c *Cli) getResult(
 
 	var n int
 
-	pollFds := []unix.PollFd{{Fd: int32(c.pollFd), Events: unix.POLLIN}}
+	pollFds := []unix.PollFd{{Fd: int32(c.pollFd), Events: unix.POLLIN}} //nolint: gosec
 
 	for {
 		if ctx.Err() != nil {

--- a/constants/session.go
+++ b/constants/session.go
@@ -6,6 +6,7 @@ const (
 	// DefaultReadDelayMaxNs is the default maximum read delay as set in the zig bits.
 	DefaultReadDelayMaxNs uint64 = 25_000_000
 
-	// ReadyFDPollTimeoutNs is the timeout to use when polling the driver's ready signal.
-	ReadyFDPollTimeoutNs int64 = 100_000_000
+	// ReadyFDPollTimeoutMs is the timeout in milliseconds to use when polling the driver's
+	// ready signal.
+	ReadyFDPollTimeoutMs int = 100
 )

--- a/netconf/netconf.go
+++ b/netconf/netconf.go
@@ -314,23 +314,18 @@ func (n *Netconf) getResult(
 
 	var _n int
 
-	readFdSet := &unix.FdSet{}
+	pollFds := []unix.PollFd{{Fd: int32(n.pollFd), Events: unix.POLLIN}}
 
 	for {
 		if ctx.Err() != nil {
 			return nil, ctx.Err()
 		}
 
-		// select can modify both the fd set and the timeout (on some platforms), so we must
-		// re-initialize them both every iteration.
-		tv := unix.NsecToTimeval(scrapligoconstants.ReadyFDPollTimeoutNs)
-
-		readFdSet.Zero()
-		readFdSet.Set(n.pollFd)
+		pollFds[0].Revents = 0
 
 		var err error
 
-		_n, err = unix.Select(n.pollFd+1, readFdSet, nil, nil, &tv)
+		_n, err = unix.Poll(pollFds, scrapligoconstants.ReadyFDPollTimeoutMs)
 		if err != nil {
 			if errors.Is(err, unix.EINTR) {
 				// python automagically handles interrupts i guess go doesnt, so just act like
@@ -342,6 +337,13 @@ func (n *Netconf) getResult(
 		}
 
 		if _n > 0 {
+			if pollFds[0].Revents&unix.POLLNVAL != 0 {
+				return nil, scrapligoerrors.NewFfiError(
+					"waiting on operation ready signal",
+					unix.EBADF,
+				)
+			}
+
 			break
 		}
 	}

--- a/netconf/netconf.go
+++ b/netconf/netconf.go
@@ -314,7 +314,7 @@ func (n *Netconf) getResult(
 
 	var _n int
 
-	pollFds := []unix.PollFd{{Fd: int32(n.pollFd), Events: unix.POLLIN}}
+	pollFds := []unix.PollFd{{Fd: int32(n.pollFd), Events: unix.POLLIN}} //nolint: gosec
 
 	for {
 		if ctx.Err() != nil {

